dotnet/src/SemanticKernel.Core/Functions/KernelFunctionFromMethod.cs

```diff
@@ -597,39 +591,42 @@ private static void TrackUniqueParameterType(ref bool hasParameterType, MethodIn
                 targetType = Nullable.GetUnderlyingType(targetType)!;
             }
 
-            // For enums, delegate to Enum.Parse, special-casing null if it was actually Nullable<EnumType>.
-            if (targetType.IsEnum)
+            // Finally, look up and use a type converter. Again, special-case null if it was actually Nullable<T>.
+            if (TypeConverterFactory.GetTypeConverter(targetType) is TypeConverter converter)
             {
                 return (input, cultureInfo) =>
                 {
-                    if (wasNullable && input is null)
+                    // This if block returns null if the target ValueType is nullable, or if the target type is a ReferenceType, which is inherently nullable.
+                    // This prevents null from being handled by converters below, which may fail when converting from nulls or to the target type from nulls.
+                    if (input is null && (wasNullable || !targetType.IsValueType))
```

Comment:
Could we move the IsValueType check out to where wasNullable is set? That way we're not evaluating it on every invocation and it simplifies the condition here, e.g.
```C#
bool wasNullable = !targetType.IsValueType;
if (!wasNullable && targetType.IsGenericType && targetType.GetGenericTypeDefinition() == typeof(Nullable<>))
...
```

Reply:
It worked, thanks!